### PR TITLE
Fixes migration for old score bug

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/3162

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
defined in https://github.com/brave/brave-browser/issues/3162

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat_helper.h [hunk 1]
```diff
@@ -226,6 +226,7 @@ namespace braveledger_bat_helper {
     bool allow_videos_ = true;
     std::map<std::string, REPORT_BALANCE_ST> monthly_balances_;
     std::map<std::string, double> recurring_donation_;
+    bool migrate_score = false;
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat_helper.h]: we don't need to migrate new users

File: vendor/bat-native-ledger/src/bat_publishers.cc [hunk 2]
```diff
@@ -532,11 +541,16 @@ void BatPublishers::synopsisNormalizerInternal(
...
     // https://github.com/brave/brave-browser/issues/3134
-    if (list[i].score < -100000) {
+    if (GetMigrateScore()) {
       list[i].score = concaveScore(list[i].duration);
     }
     totalScores += list[i].score;
   }
+
+  if (GetMigrateScore()) {
+    SetMigrateScore(false);
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat_publishers.cc]: we only want migrate them once

File: vendor/bat-native-ledger/src/bat_publishers.cc [hunk 3]
```diff
@@ -660,9 +674,8 @@ bool BatPublishers::isEligibleForContribution(const ledger::PublisherInfo& info)
   if (isExcluded(info.id, info.excluded) || (!state_->allow_non_verified_ && !isVerified(info.id)))
     return false;
 
-  return info.score > 0 &&
-    info.duration >= state_->min_publisher_duration_ &&
-    info.visits >= state_->min_visits_;
+  return info.duration >= state_->min_publisher_duration_ &&
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat_publishers.cc]: we need to remove this check for score as old versions had negative values, which was now fixed

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat_publishers.cc]: I would keep it as it. This code was not changed and would like to minimize  changes as much as possible as this goes into release.

File: vendor/bat-native-ledger/src/bat_helper.cc [hunk 4]
```diff
@@ -661,6 +662,7 @@ static bool ignore_ = false;
           monthly_balances_.insert(std::make_pair(itr->name.GetString(), r));
         }
       }
+
```
Comment (jasonrsadler) [inline vendor/bat-native-ledger/src/bat_helper.cc]: Remove this line

**Pull request merged**